source/common/upstream/strict_dns_cluster.cc

```diff
@@ -91,10 +91,12 @@ StrictDnsClusterImpl::ResolveTarget::ResolveTarget(
     StrictDnsClusterImpl& parent, Event::Dispatcher& dispatcher, const std::string& url,
     const envoy::config::endpoint::v3::LocalityLbEndpoints& locality_lb_endpoint,
     const envoy::config::endpoint::v3::LbEndpoint& lb_endpoint)
-    : parent_(parent), dns_address_(Network::Utility::hostFromTcpUrl(url)),
+    : parent_(parent), locality_lb_endpoints_(locality_lb_endpoint), lb_endpoint_(lb_endpoint),
+      dns_address_(Network::Utility::hostFromTcpUrl(url)),
+      hostname_(lb_endpoint_.endpoint().hostname().empty() ? dns_address_
+                                                           : lb_endpoint_.endpoint().hostname()),
```

Comment:
Not sure if it space matters here but I think we could get away with `hostname_` being a const ref since it's referencing either dns_address or `lb_endpoint_`?

Reply:
Yeah I think we could, though I'm not sure it's worth a potential bug in the future.

Reply:
(Also if you feel strongly about this I'm happy to change it, so just let me know and/or approve :))

Reply:
Leaving as is is fine I think, I agree that a ref is more likely to cause bugs in the future.